openpype/hosts/blender/plugins/create/create_render.py

```diff
@@ -0,0 +1,302 @@
+        file_name, _ = os.path.splitext(file_name)
+
+        project = get_current_project_name()
+        settings = get_project_settings(project)
+
+        render_folder = self.get_default_render_folder(settings)
+        aov_sep = self.get_aov_separator(settings)
+        ext = self.get_image_format(settings)
+        multilayer = self.get_multilayer(settings)
+
+        aov_list, custom_passes = self.set_render_passes(settings)
+
+        output_path = os.path.join(file_path, render_folder, file_name)
+
+        render_product = self.get_render_product(output_path, name)
+        aov_file_product = self.set_node_tree(
+            output_path, name, aov_sep, ext, multilayer)
+
+        # We set the render path, the format and the camera
+        bpy.context.scene.render.filepath = render_product
+        self.set_render_format(ext, multilayer)
+
+        render_settings = {
+            "render_folder": render_folder,
+            "aov_separator": aov_sep,
+            "image_format": ext,
+            "multilayer_exr": multilayer,
+            "aov_list": aov_list,
+            "custom_passes": custom_passes,
+            "render_product": render_product,
+            "aov_file_product": aov_file_product,
+        }
+
+        self.imprint_render_settings(asset_group, render_settings)
+
+    def process(self):
+        # Get Instance Container or create it if it does not exist
+        instances = bpy.data.collections.get(AVALON_INSTANCES)
+        if not instances:
+            instances = bpy.data.collections.new(name=AVALON_INSTANCES)
+            bpy.context.scene.collection.children.link(instances)
+
+        # Create instance object
+        asset = self.data["asset"]
+        subset = self.data["subset"]
+        name = plugin.asset_name(asset, subset)
+        asset_group = bpy.data.collections.new(name=name)
+
+        try:
+            instances.children.link(asset_group)
+            self.data['task'] = get_current_task_name()
+            lib.imprint(asset_group, self.data)
+
+            self.prepare_rendering(asset_group, name)
+        except Exception:
+            # Remove the instance if there was an error
+            bpy.data.collections.remove(asset_group)
+            raise
+
+        # TODO: this is undesiderable, but it's the only way to be sure that
+        # the file is saved before the render starts.
+        # Blender, by design, doesn't set the file as dirty if modifications
+        # happen by script. So, when creating the instance and setting the
+        # render settings, the file is not marked as dirty. This means that
+        # there is the risk of sending to deadline a file without the right
+        # settings. Even the validator to check that the file is saved will
+        # detect the file as saved, even if it isn't. The only solution for
+        # now it is to force the file to be saved.
+        bpy.ops.wm.save_as_mainfile(filepath=bpy.data.filepath)
```

Comment:
Would it work to just force `bpy.data.is_dirty = True`?

Reply:
Yeah, I've tried everything, but I found that same post ([and others with Blender developers comments saying that it is like that by design](https://devtalk.blender.org/t/how-to-get-bpy-data-is-dirty-updating-when-changes-are-done-by-script/4852/2)), so I don't think there's any other way, for now :/